### PR TITLE
SISRP-15417 Handle GL5 API wrapper format from Hub

### PR DESCRIPTION
**Pull request by paulkerschen**
DO NOT MERGE until @raydavis and Ben Hubbard have signed off.

https://jira.berkeley.edu/browse/SISRP-15417

We are very lucky that #4586 has already centralized Hub response parsing, so that only a small change in code was needed. The large changes are in the fixtures.

Backwards compatibility is attempted. While this buys us more flexibility in how we deploy, it of course adds complication and risk.

### CODE REVIEW

File: app/models/hub_edos/response_handler.rb [hunk 1]
```diff
@@ -3,10 +3,14 @@ module ResponseHandler
 
     def get_students(response)
       return [] unless response.is_a? Hash
-      container = response['studentResponse']
-      # Verify that target element is capable of being iterated. And yes, the structure really is this weird.
-      return [] unless container.present? && container['students'].present? && container['students']['students'].respond_to?(:each)
-      container['students']['students']
+      # We are temporarily handling both GL4 and GL5 wrapper formats for backwards compatibility.
```
Comment (christianvuerings) [inline app/models/hub_edos/response_handler.rb]: @paulkerschen thank you!

**Comment by christianvuerings** [comment]
Lgtm but waiting for review from @raydavis & green travis build.

**Comment by paulkerschen** [comment]
Bamboo build at https://bamboo.ets.berkeley.edu/bamboo/browse/CPB-CPT276-1/ . HubEdos::WorkExperience is a longstanding known failure, because no one except @davidlimcheng is known to have work experience data in the Hub. All other Hub-dependent tests seems to have passed OK.

**Comment by paulkerschen** [comment]
Oops, just found a `pp` in a spec. Amended; Travis is rebuilding.

File: app/models/hub_edos/response_handler.rb [hunk 2]
```diff
@@ -3,10 +3,14 @@ module ResponseHandler
...
+      wrapper_keys = if response['apiResponse']
+                       %w(apiResponse response any students)
+                     else
+                       %w(studentResponse students students)
+                     end
+      students = wrapper_keys.inject(response) { |feed, key| (feed.is_a?(Hash) && feed[key]) || [] }
```
Comment (johncrossman) [inline app/models/hub_edos/response_handler.rb]: All looks good, @paulkerschen  One suggestion: your added comment (above) should tell us which response is GL4 and which is GL5.

Comment (paulkerschen) [inline app/models/hub_edos/response_handler.rb]: Thanks @johncrossman; comment added.

**Comment by johncrossman** [comment]
Thanks, @paulkerschen  I say +1. Your turn, @raydavis

**Comment by raydavis** [comment]
Yay, @johncrossman did exactly the localized unwrapper we were hoping to find!

**Pull request merged**